[C#] extract concepts of v2 isbn-verifier exercise (#537)
exercism/v3: languages/csharp/reference/exercise-concepts/isbn-verifier.md

[Example implementation](https://github.com/exercism/csharp/blob/master/exercises/isbn-verifier/Example.cs)

 ## General
 - functions: used as the main entry point to this exercise
 - static functions: used as the main entry point to this exercise

>  - function arguements: input string is passed as an arguement

Reviewer: ```suggestion
 - function arguments: input string is passed as an argument
```

 - return values: returning value from method
 - visibility: test class and test methods are `public`
 - imports: importing types with using statements
 - namespaces: knowing where to find Regular Expression methods
 - regex: use regex to implement the checker
 - strings: used to store the isbn numbers
 - for loop: iterate over the numbers to check if valid 

>  - modulus: checking return value has no remainder

Reviewer: I'd argue this is just another math operator :)

 - conditionals: used for checking values

>  - collasce: combine coniditonals to check for isbn

Reviewer: ```suggestion
 - coalesce operator: combine conditonals to check for isbn
```

Author: I changed this to ternary `?` operator, because we're not doing any coalesce `??`

[Example implementation](https://github.com/exercism/csharp/blob/master/exercises/isbn-verifier/Example.cs)

 ## General
 - functions: used as the main entry point to this exercise
 - static functions: used as the main entry point to this exercise

>  - regex: use regex to implment the checker

Reviewer 2: ```suggestion
 - regex: use regex to implement the checker
```

>  - arrays: used to store the isbn numbers

Reviewer 2: I'd personally call this `strings` even though it can technically be treated as array of characters

Reviewer: I agree with the `strings` suggestion

>  - for loop: itterate over the numbers to check if valid 

Reviewer 2: ```suggestion
 - for loop: iterate over the numbers to check if valid 
```
